hphp/hack/src/parser/full_fidelity_ast.ml

```diff
@@ -3671,6 +3671,7 @@ let defensive_program
       ~elaborate_namespaces
       ~keep_errors:(keep_errors || (not fail_open))
       ~parser_options
+      ~enable_hh_syntax:true
```

Comment:
Why has this changed? Can you use `!is_ide_mode` here too?

Reply:
ah woops. I initially did this by keying on force_hh_syntax instead of the new global flag - but that turned out to be a bad idea given that hh_server is used for non-IDE queries too and we need to preserve the meaning